test/common/dns.js

```diff
@@ -0,0 +1,290 @@
+              const mname = readDomainFromPacket(buffer, offset);
+              const rname = readDomainFromPacket(buffer, offset + mname.nread);
+              rr.nsname = mname.domain;
+              rr.hostmaster = rname.domain;
+              const trailerOffset = offset + mname.nread + rname.nread;
+              rr.serial = buffer.readUInt32BE(trailerOffset);
+              rr.refresh = buffer.readUInt32BE(trailerOffset + 4);
+              rr.retry = buffer.readUInt32BE(trailerOffset + 8);
+              rr.expire = buffer.readUInt32BE(trailerOffset + 12);
+              rr.minttl = buffer.readUInt32BE(trailerOffset + 16);
+
+              assert.strictEqual(trailerOffset + 20, dataLength);
+              break;
+            }
+          default:
+            throw new Error(`Unknown RR type ${rr.type}`);
+        }
+        offset += dataLength;
+      }
+
+      parsed[sectionName].push(rr);
+
+      assert.ok(offset <= buffer.length);
+    }
+  }
+
+  assert.strictEqual(offset, buffer.length);
+  return parsed;
+}
+
+function writeIPv6(ip) {
+  const parts = ip.replace(/^:|:$/g, '').split(/:/g);
+  const buf = Buffer.alloc(16);
+
+  let offset = 0;
+  for (const part of parts) {
+    if (part === '') {
+      offset += 16 - 2 * (parts.length - 1);
+    } else {
+      buf.writeUInt16BE(parseInt(part, 16), offset);
+      offset += 2;
+    }
+  }
+
+  return buf;
+}
+
+function writeDomainName(domain) {
+  return Buffer.concat(domain.split(/\./g).map((label) => {
+    assert(label.length < 64);
+    return Buffer.concat([
+      Buffer.from([label.length]),
+      Buffer.from(label, 'ascii')
+    ]);
+  }).concat([Buffer.alloc(1)]));
+}
+
+function writeDNSPacket(parsed) {
+  const buffers = [];
+
+  buffers.push(new Uint16Array([
+    parsed.id,
+     // default to “standard response”
+    parsed.flags === undefined ? 0x8180 : parsed.flags,
+    parsed.questions && parsed.questions.length,
+    parsed.answers && parsed.answers.length,
+    parsed.authorityAnswers && parsed.authorityAnswers.length,
+    parsed.additionalRecords && parsed.additionalRecords.length,
+  ]));
+
+  for (const q of parsed.questions) {
+    assert(types[q.type]);
+    buffers.push(writeDomainName(q.domain));
+    buffers.push(new Uint16Array([
+      types[q.type],
+      q.cls === undefined ? classes.IN : q.cls
+    ]));
+  }
+
+  for (const rr of [].concat(parsed.answers,
+                             parsed.authorityAnswers,
+                             parsed.additionalRecords)) {
+    if (!rr) continue;
+
+    assert(types[rr.type]);
+    buffers.push(writeDomainName(rr.domain));
+    buffers.push(new Uint16Array([
+      types[rr.type],
+      rr.cls === undefined ? classes.IN : rr.cls
+    ]));
+    buffers.push(new Int32Array([rr.ttl]));
+
+    const rdLengthBuf = new Uint16Array(1);
+    buffers.push(rdLengthBuf);
+
+    switch (rr.type) {
+      case 'A':
+        rdLengthBuf[0] = 4;
+        buffers.push(new Uint8Array(rr.address.split(/\./g)));
```

Comment:
There are a couple uses of `split()` on single characters in this PR. Would the string version work (`split('.')`)? It seems more readable.